server/apps/distributed-app/src/main/java/org/apache/james/CassandraRabbitMQJamesServerMain.java

```diff
@@ -205,6 +207,20 @@ public static GuiceJamesServer createServer(CassandraRabbitMQJamesConfiguration
             .combineWith(chooseJmapModule(configuration));
     }
 
+    private static Module chooseMailQueue(CassandraRabbitMQJamesConfiguration configuration) {
+        switch (configuration.getMailQueueChoice()) {
+            case PULSAR:
+                return new PulsarQueueModule();
```

Comment:
IMO we should write an Integration test class with Pulsar mail queue that implements some basic James contract test e.g `JamesServerContract` to make sure James can run normally at very basic operations.

Reply:
Let me try it